src/dropdown/dropdown.spec.ts

```diff
@@ -515,6 +515,42 @@ describe('ngb-dropdown-toggle', () => {
       expect(getDropdownEl(compiled)).toHaveCssClass('dropup');
     });
   });
+
+  it('should correctly apply auto placement', () => {
+    const html = `
+      <div ngbDropdown placement="auto">
+          <div ngbDropdownMenu>
+            <a class="dropdown-item">dropDown item</a>
+            <a class="dropdown-item">dropDown item</a>
+          </div>
+      </div>`;
+
+    const fixture = createTestComponent(html);
+    const compiled = fixture.nativeElement;
+
+    const dropDownEle = getDropdownEl(compiled);
+
+    expect(dropDownEle).toHaveCssClass('dropdown');
+    expect(dropDownEle.getAttribute('ng-reflect-placement')).toEqual('auto');
```

Comment:
We should not use `ng-reflect-` attributes in tests as it is impl-specific and can change at any moment (not to mention the fact that it is not there in production mode).

TBH I'm not sure what you are trying to assert with this test so it is hard for me to suggest changes...